plasma_framework/test/src/utils/BondSize.test.js

```diff
@@ -6,87 +6,195 @@ contract('BondSize', () => {
+            });
+
+            it('should be able to update the bond to lowerBoundDivisor of its current value', async () => {
+                const newBondSize = new BN(this.initialBondSize).divn(this.lowerBoundDivisor);
+                await this.contract.updateBondSize(newBondSize);
+            });
+
+            it('should fail to update bond to less than lowerBoundDivisor of its current value', async () => {
+                const newBondSize = new BN(this.initialBondSize).divn(this.lowerBoundDivisor).subn(1);
+                await expectRevert(
+                    this.contract.updateBondSize(newBondSize),
+                    'Bond size is too low',
+                );
+            });
+
+            it('should not update the actual bond value until after the waiting period', async () => {
+                const newBondSize = new BN(this.initialBondSize).muln(2);
+                await this.contract.updateBondSize(newBondSize);
+
+                await time.increase(WAITING_PERIOD.sub(time.duration.seconds(5)));
+                const bondSize = await this.contract.bondSize();
+                expect(bondSize).to.be.bignumber.equal(this.initialBondSize);
+            });
+
+            it('should update the actual bond value after the waiting period', async () => {
+                const newBondSize = new BN(this.initialBondSize).muln(2);
+                await this.contract.updateBondSize(newBondSize);
+
+                await time.increase(WAITING_PERIOD);
+                const bondSize = await this.contract.bondSize();
+                expect(bondSize).to.be.bignumber.equal(newBondSize);
+            });
+
+            it('should update the actual bond value after the waiting period', async () => {
+                const newBondSize = new BN(this.initialBondSize).muln(2);
+                await this.contract.updateBondSize(newBondSize);
+
+                // Wait half the waiting period
+                await time.increase(HALF_WAITING_PERIOD);
+
+                // Update again while the first update is in progress.
+                const secondNewBondSize = new BN(this.initialBondSize).muln(1.5);
+                await this.contract.updateBondSize(secondNewBondSize);
+
+                // Wait half the waiting period again.
+                await time.increase(HALF_WAITING_PERIOD);
+                // Even though the first update's waiting period is over, the second
+                // update is in progress so the bond size should not have changed yet.
+                let bondSize = await this.contract.bondSize();
+                expect(bondSize).to.be.bignumber.equal(this.initialBondSize);
+
+                // Wait the remaining waiting period
+                await time.increase(HALF_WAITING_PERIOD);
+                bondSize = await this.contract.bondSize();
+                expect(bondSize).to.be.bignumber.equal(secondNewBondSize);
+            });
+        });
+
+        describe('with boundary size of numbers', () => {
```

Comment:
@thec00n here is the new tests for boundary

Reply:
I did not appreciate https://github.com/omisego/plasma-contracts/blob/11b3bae4926dabc491dfbce511207b2f2798ba0f/plasma_framework/contracts/src/exits/utils/BondSize.sol#L75 in the beginning but I think it's actually the best solution in this case to ensure that the check does not overflow and scenario where we get stuck with a new bond size.